app/middleware/auth_redirect_middleware.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Rack
```

Comment:
You have invalid naming. According to your directory and file name, it should be `Middleware::AuthRedirectMiddleware`.
Please check zeitwerk gem documentation for more examples.
P.S. custom code like this middleware is usually located in the `lib` folder, not in the `middleware`

Reply:
ok

Reply:
My middleware must be located in root of lib/ directory?

Reply:
yes, in the root folder
`lib/rack/auth_redirect` if you want to keep the naming Lib::Rack::AuthRedirect